source/PlayerInfo.cpp

```diff
@@ -841,15 +841,32 @@ void PlayerInfo::AddShip(const shared_ptr<Ship> &ship)
 
 
 
-// Buy a ship of the given model, and give it the given name.
-void PlayerInfo::BuyShip(const Ship *model, const string &name)
+// Charge the player the cost of a ship and give it the specified name:
+void PlayerInfo::BuyShip(const Ship *model, const std::string &name)
+{
+	ReceiveShip(model, name, DEFAULT_COST, true);
+}
+
+
+
+// Receive a ship as a gift
+shared_ptr<Ship> PlayerInfo::GiftShip(const Ship *model, const std::string &name, bool record_depriciation)
+{
+	return ReceiveShip(model, name, 0, record_depriciation);
+}
+
+
+
+// Purchase or receive a gift of a ship of the given model and name
+shared_ptr<Ship> PlayerInfo::ReceiveShip(const Ship *model, const string &name, int64_t cost, bool record_depriciation)
```

Comment:
Why does ReceiveShip return a pointer to the ship? Under what circumstances would this ever return anything aside from the ship that was just given to it? The only instance in which a nullptr would be returned is if the cost was greater than the player had, but gifted ships have no cost. (A nullptr could also be returned if the model doesn't exist, but checking if the model exists is already done by DoGiftShip in MissionAction.)